lib/zone.ts

```diff
@@ -1387,8 +1387,8 @@ const Zone: ZoneType = (function(global: any) {
     // if this is valid, means not undefined
     // and not global (non-strict mode call Error())
     // and it is an error not an unrelated object
-    if (this && this !== global && this instanceof Error
-      && Object.getPrototypeOf(this) !== NativeError.prototype && assignAll(this, error)) {
+    if (this && this !== global && this instanceof Error &&
```

Comment:
I was thinking about this some more, and I think the best way to do this would be to place this guard at the top of the function

```
if (! this instanceof ZoneAwareError) {
  return ZoneAwareError.apply(Object.create(ZoneAwareError), arguments);
}
```

This way we force the `new` and only have one code path. What do you think?